### ISSUE TITLE
Question about the batch normalization across GPUs

### ISSUE THREAD

**Issue by arrufat** [davisking/dlib#2404]
I recently noticed that PyTorch has 2 kinds of normalization, the standard one, and the `SyncBatchNorm`, which, from my understanding, instead of computing the batch normalization on each GPU separately, it does it across GPUs.

I wondered which kind of batch normalization dlib uses. I had a brief look at the code, but I couldn't find any information about it. Maybe we could add this to the `bn_` layer documentation?

Thanks :)

**Comment by arrufat** [comment]
From reading the trainer code, it seems like each GPU does its own stuff then the tensors are averaged at the end, right?
So, for batch normalization, there is nothing special going on, we are averaging the batch normalization parameters from all devices, right?

To make sure I understand what's going on
- if we set a batch size _B_ across _N_ devices
- each device works "independently" with batch size _B_/_N_
- at the end of the forward pass, we average all the parameters of the networks from the _N_ devices

So, for the batch normalization it works by normalizing a batch size of _B_/_N_ and then averaging the _N_ tensors, which is not the same as computing the batch normalization with a batch size of _B_, right?

**Comment by davisking** [comment]
Right, it's the non-sync version.

**Comment by arrufat** [comment]
Thank you!
